src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/filter/EJSONWrapper.java

```diff
@@ -0,0 +1,86 @@
+package io.stargate.sgv2.jsonapi.api.model.command.clause.filter;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import java.nio.ByteBuffer;
+import java.util.Objects;
+
+/**
+ * Intermediate type for further processing: used with {@link JsonLiteral} and {@link
+ * JsonType#EJSON_WRAPPER} to store special types in JSON (e.g. binary data). Currently only one
+ * type is supported but this can be extended in the future.
+ */
+public class EJSONWrapper {
+  // Actual DRY keys
+  public static final String $BINARY = "$binary";
+
+  public enum EJSONType {
+    BINARY($BINARY);
+
+    private final String key;
+
+    EJSONType(String key) {
+      this.key = key;
+    }
+
+    public static EJSONType fromKey(String key) {
+      // With very few entries just do this: can build lookup Map if needed
+      // in future
+      return switch (key) {
+        case $BINARY -> BINARY;
+        default -> null;
```

Comment:
There's a reason I am not throwing an exception tho: when we get this call, it is not guaranteed it must be an EJSON value -- it could just be legit Map with funny single entry:

```
{ "$abc" : 123 }
```

so we only want to construct `EJSONWrapper` where we do recognize likely type.
Alternatively we could have "EJSONCandidate" with weak typing, name as String, and resolve later, but seemed like better way to bind early.
But even in this case, when supporting `map` types, may need to "undo" `EJSONWrapper` for `map` type -- information that is only available later, when we have Schema information. And not yet during shredding.